### PR TITLE
Update network validation script

### PR DESCRIPTION
**Pull request by PM-Darshan**
**Testing**

1. Errors shown when no rules are added for Security Lists.
    Screenshot 2024-05-23 at 7 10 17 PM

2. No errors, when all the required rules are added for Security Lists.
     Screenshot 2024-05-23 at 7 16 41 PM Screenshot 2024-05-23 at 7 16 51 PM Screenshot 2024-05-23 at 7 17 00 PM Screenshot 2024-05-23 at 10 58 56 PM

3. Errors shown when no rules are added for Network Security Groups.
    Screenshot 2024-05-23 at 7 27 44 PM

4. No errors, when all the required rules are added for Network Security Groups.
     Screenshot 2024-05-23 at 8 00 16 PM Screenshot 2024-05-23 at 7 31 25 PM Screenshot 2024-05-23 at 7 32 18 PM Screenshot 2024-05-23 at 7 33 04 PM Screenshot 2024-05-23 at 7 33 39 PM

### CODE REVIEW

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 1]
```diff
@@ -802,6 +823,21 @@ then
   then
     echo "WARNING: Exposing the WebLogic administrator port [${ADMIN_HTTPS_PORT}] in the subnet [{$WLS_SUBNET_OCID}] to the internet [${ALL_IPS}] allows any user to access the WebLogic console, which is not a recommended practice. Ensure that only a specific CIDR range can access the WebLogic console. ${NETWORK_VALIDATION_MSG}"
   fi
+
+  # Check if Admin Console HTTPS Port is open for access by WLS subnet CIDR for secure mode
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:827]: Please change "Admin Console HTTPS Port" to "Administration Port". This is the name of the port in WLS documentation so we should keep the name consistent.

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 2]
```diff
@@ -859,6 +897,21 @@ then
   then
     echo "WARNING: Exposing the WebLogic administrator port [${ADMIN_HTTPS_PORT}] in the Admin Server NSG [{$ADMIN_SRV_NSG_OCID}] to the internet [${ALL_IPS}] allows any user to access the WebLogic console, which is not a recommended practice. Ensure that only a specific CIDR range can access the WebLogic console. ${NETWORK_VALIDATION_MSG}"
   fi
+
+  # Check if Admin Console HTTPS Port is open for access by WLS subnet CIDR in Managed Server NSG for secure mode
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:901]: Please change "Admin Console HTTPS Port" to "Administration Port". This is the name of the port in WLS documentation so we should keep the name consistent.

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 3]
```diff
@@ -1019,6 +1072,39 @@ then
         fi
       fi
     fi
+
+    # In secure production mode, check if Check if bastion Host IP CIDR or Bastion Subnet CIDR has access to Admin HTTPS port on WLS subnet or Admin Server NSG
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:1076]: Please change "Admin HTTPS Port" to "Administration Port". This is the name of the port in WLS documentation so we should keep the name consistent.

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 4]
```diff
@@ -658,6 +662,21 @@ if ${debug}; then set -x ; fi
 # This way you can catch the error in case mysqldump fails in `mysqldump |gzip`, for example.
 set -o pipefail
 
+# Convert to lowercase for case insensitive check
+secure_mode=$(echo "$SECURE_MODE" | tr '[:upper:]' '[:lower:]')
+
+# Check if script needs  to be validated for secure production mode
+if [ "$secure_mode" = "true" ]; then
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:669]: It seems like this switch between IDCS port and managed server port would be useful for non-production mode as well. Seems like an oversight in the existing script to me. @Mahuwa-Barman , do you agree?

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 5]
```diff
@@ -658,6 +662,21 @@ if ${debug}; then set -x ; fi
...
+    # Changing the required ports default value for secure production mode
+    if [[ -n ${IDCS_PORT} ]]
+    then
+      WLS_LB_PORT=${IDCS_PORT}
+    else
+      WLS_LB_PORT=7004
+    fi
+    ADMIN_HTTPS_PORT=9002
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:677]: This shouldn't be hard coded. A customer is able to set a different Administration Port in secured production mode.

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 6]
```diff
@@ -658,6 +662,21 @@ if ${debug}; then set -x ; fi
...
+secure_mode=$(echo "$SECURE_MODE" | tr '[:upper:]' '[:lower:]')
+
+# Check if script needs  to be validated for secure production mode
+if [ "$secure_mode" = "true" ]; then
+    # Changing the required ports default value for secure production mode
+    if [[ -n ${IDCS_PORT} ]]
+    then
+      WLS_LB_PORT=${IDCS_PORT}
+    else
+      WLS_LB_PORT=7004
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:675]: This shouldn't be hard coded. I believe that a customer should be able to change this port in secured production mode.

File: terraform/modules/network-validator/scripts/network_validation.sh (hunk 3 above)
Comment (Mahuwa-Barman) [reply terraform/modules/network-validator/scripts/network_validation.sh:1076]: Please change secure production mode -> secured production mode

File: terraform/modules/network-validator/scripts/network_validation.sh (hunk 4 above)
Comment (Mahuwa-Barman) [reply terraform/modules/network-validator/scripts/network_validation.sh:669]: Yes

**Comment by PM-Darshan** [comment]
**Updated Testing**

Now, when https_port and wls_lb_port are provided, it would check for that. If not provided, it by default checks for 9002 and 7004. Also the switch between IDCS and Managed Server port supports both secured and non-secured production mode.
Screenshot 2024-05-27 at 12 14 51 PM

**Review by telake — APPROVED** [review]
Approving subject to adding the comment by the setting of the default T3_PORT variable in secured production mode.

**Pull request merged**